coldfront/core/utils/management/commands/export_data.py

```diff
@@ -452,7 +471,95 @@ def handle_survey_responses(self, *args, **kwargs):
                 surveys.append({
                     'project_name': project.name,
                     'project_title': project.title,
-                    'survey_responses': survey
+                    'new_project_survey_responses': survey
+                })
+
+            surveys = list(sorted(surveys, key=lambda x: x['project_name'], reverse=True))
+            self.to_json(surveys,
+                         output=kwargs.get('stdout', stdout),
+                         error=kwargs.get('stderr', stderr))
+    
+    def handle_renewal_survey_responses(self, *args, **kwargs):
+        format = kwargs['format']
+        allowance_type = kwargs['allowance_type']
+        allocation_requests = AllocationRenewalRequest.objects.all()
+        allocation_period = kwargs['allocation_period']
+
+
+        if allowance_type:
+            allocation_requests = allocation_requests.filter(
+                post_project__name__istartswith=allowance_type)
+            
+        if allocation_period:
+             allocation_requests = allocation_requests.filter(
+                 allocation_period__name=allocation_period)
+
+        _surveys = list(allocation_requests.values_list('renewal_survey_answers', flat=True))
+        if {} in _surveys:
+            raise Exception("This allocation period does not have an associated survey.")
+        surveys = []
+        # Create dict of multiple choice fields to replace field IDs with text. ID : text
+        multiple_choice_fields = {}
+        form = ProjectRenewalSurveyForm()
+        for k, v in form.fields.items():
+            # Only ChoiceField or MultipleChoiceField (in this specific survey form) have choices 
+            if (isinstance(v, forms.MultipleChoiceField)) or (isinstance(v, forms.ChoiceField)):
+                multiple_choice_fields[k] = {k: v for k, v in form.fields[k].choices}
+
+        if format == 'csv':
+            for allocation_request, survey in zip(allocation_requests, _surveys):
+                for question, answer in survey.items():
+                    if question in multiple_choice_fields.keys():
+                        sub_map = multiple_choice_fields[question]
+                        if (isinstance(answer, list)):
+                            # Multiple choice, array
+                            survey[question] = [sub_map.get(i,i) for i in answer]
+                        else:
+                            # Single choice replacement
+                            survey[question] = sub_map[answer]
```

Comment:
```
  File "/var/www/coldfront_app/coldfront/coldfront/core/utils/management/commands/export_data.py", line 525, in handle_renewal_survey_responses
    survey[question] = sub_map[answer]
KeyError: ''
```

Didn't investigate deeply, but I think this is happening because the multiple choice fields (mentioned below) are not required.

Reply:
Fixed by adding a check to skip swapping for empty answers